bin/api_diff.ml

```diff
@@ -4,14 +4,19 @@ let run (`Ref_cmi reference) (`Current_cmi current) =
   let current = Cmi_format.read_cmi current in
   let reference = Cmi_format.read_cmi reference in
   let typing_env = Env.empty in
-  let coercion () =
+  let coercion1 () =
     Includemod.signatures typing_env ~mark:Mark_both reference.cmi_sign
       current.cmi_sign
   in
-  match coercion () with
-  | Tcoerce_none -> Printf.printf "API unchanged!\n"
-  | _ -> Printf.printf "API changed!\n"
-  | exception Includemod.Error _ -> Printf.printf "API changed!\n"
+  let coercion2 () =
+    Includemod.signatures typing_env ~mark:Mark_both current.cmi_sign
+      reference.cmi_sign
+  in
+  try
+    match (coercion1 (), coercion2 ()) with
+    | Tcoerce_none, Tcoerce_none -> Printf.printf "API unchanged!\n"
+    | _, _ -> Printf.printf "API changed!\n"
+  with Includemod.Error _ -> Printf.printf "API changed!\n"
```

Comment:
Let's keep using the case analysis to catch the exception, as we are currently:

```suggestion
  match (coercion1 (), coercion2 ()) with
  | Tcoerce_none, Tcoerce_none -> Printf.printf "API unchanged!\n"
  | _, _ -> Printf.printf "API changed!\n"
  | exception Includemod.Error _ -> Printf.printf "API changed!\n"
```

Reply:
Yes. I was trying to do it like this 
```
match (coercion1 (), coercion2 ()) with
  | Tcoerce_none, Tcoerce_none -> Printf.printf "API unchanged!\n"
  | _, _ -> Printf.printf "API changed!\n"
  | Includemod.Error _, Includemod.Error _ -> Printf.printf "API changed!\n"
```
and the build was failing. Now it is fine. Thanks. I have pushed the commit.

Reply:
Ah yea! I can see how that would make sense! As you probably see from this change now, [exception patterns](https://v2.ocaml.org/manual/patterns.html#sss:exception-match) are basically just syntax sugar for wrapping a `match` expression in a `try`, and "exception cases" match against exceptions that arise when evaluating the matched expression, and so are not related to the structure of the matched expression themself.